src/python/py/models/builder.py

```diff
@@ -413,16 +415,74 @@ def save_model(self, out_dir):
         )
 
     def to_int4(self, model):
-        quant = MatMul4BitsQuantizer(
+        if (self.quant_provider == "nvidia_awq"):
+            # Nvidia ModelOpt only accepts an ONNX model path as input so process is
+            # hf --> fp16 model.onnx file (store at temp) --> pass fp16 model.onnx to modelopt
+            import tempfile
+            from onnxruntime.quantization.matmul_4bits_quantizer import NVAWQWeightOnlyQuantConfig
+            out_dir = os.path.join(tempfile.gettempdir(), "temp")
+            os.makedirs(out_dir, exist_ok=True)
+
+            out_path = os.path.join(out_dir, self.filename)
```

Comment:
Since saving the ONNX model has to be done twice with `nvidia_awq`, we should move the saving logic to its own method. For example, we can move

https://github.com/microsoft/onnxruntime-genai/blob/7998f13e25f31caaba803db4197f737df99a0bb9/src/python/py/models/builder.py#L395-L413

to something like this.

```python
def to_disk(self, model, out_dir):
    # Save ONNX model with only one external data file and delete any existing duplicate copies
    out_path = os.path.join(out_dir, self.filename)
    data_path = os.path.join(out_dir, os.path.basename(out_path) + ".data")
    
    if os.path.exists(out_path):
        print(f"Overwriting {out_path}")
        os.remove(out_path)
    if os.path.exists(data_path):
        print(f"Overwriting {data_path}")
        os.remove(data_path)

    save_model(
        model,
        out_path,
        save_as_external_data=True,
        all_tensors_to_one_file=True,
        location=os.path.basename(data_path),
        size_threshold=0,
        convert_attribute=False,
    )
```

Then, we can re-use this method here as well.

Reply:
Done